Modules/Journey/tabs/QuestsByFaction/QuestsByFactions.lua

```diff
@@ -0,0 +1,540 @@
+            QuestieJourney.availableFactionExpansions[expansion.key] = expansion.label
+            table.insert(QuestieJourney.availableFactionExpansionOrder, expansion.key)
+        end
+    end
+end
+
+local function _PopulateFactionsByExpansion()
+    local referenced = _CollectReferencedFactionIds()
+
+    QuestieJourney.factionsByExpansion = QuestieJourney.factionsByExpansion or {}
+
+    for key in pairs(QuestieJourney.factionsByExpansion) do
+        QuestieJourney.factionsByExpansion[key] = nil
+    end
+
+    for _, expansion in ipairs(expansionDefinitions) do
+        QuestieJourney.factionsByExpansion[expansion.key] = {}
+    end
+
+    for factionId, introOrder in pairs(factionIntroductionOrder) do
+        if referenced[factionId] then
+            _RegisterFactionForExpansion(factionId, introOrder)
+        end
+    end
+
+    for factionId in pairs(referenced) do
+        if not factionIntroductionOrder[factionId] then
+            _EnsureFactionRegistered(factionId)
+        end
+    end
+end
+
+local function _EnsureFactionDataInitialized()
+    if factionDataInitialized then
+        return
+    end
+
+    _BuildExpansionDropdownData()
+    _PopulateFactionsByExpansion()
+
+    factionDataInitialized = true
+end
+
+function _QuestieJourney.questsByFaction:InitializeFactionData()
+    _EnsureFactionDataInitialized()
+end
+
+function _QuestieJourney.questsByFaction:InitializeFactionQuestData()
+    _EnsureFactionQuestData()
+end
+
+local function _AddQuestToFaction(factionId, questId)
+    if not factionId or not questId then
+        return
+    end
+
+    _EnsureFactionRegistered(factionId)
+
+    if not factionQuestMap[factionId] then
+        factionQuestMap[factionId] = {}
+    end
+
+    factionQuestMap[factionId][questId] = true
+end
+
+local function _IsAllianceRaceMask(raceMask)
+    if not raceMask or raceMask == QuestieDB.raceKeys.NONE then
+        return false
+    end
+
+    return bit.band(raceMask, QuestieDB.raceKeys.ALL_ALLIANCE) == raceMask
+end
+
+local function _IsHordeRaceMask(raceMask)
+    if not raceMask or raceMask == QuestieDB.raceKeys.NONE then
+        return false
+    end
+
+    return bit.band(raceMask, QuestieDB.raceKeys.ALL_HORDE) == raceMask
+end
+
+function _EnsureFactionQuestData()
+    if factionQuestMap then
+        return
+    end
+
+    factionQuestMap = {}
+
+    local queryFields = {
+        "requiredMinRep",
+        "requiredMaxRep",
+        "reputationReward",
+        "requiredRaces",
+        "requiredClasses",
+    }
+
+    for questId in pairs(QuestieDB.QuestPointers) do
+        local queryResult = QuestieDB.QueryQuest(questId, queryFields) or {}
+        local requiredMinRep = queryResult[1]
+        local requiredMaxRep = queryResult[2]
+        local reputationReward = queryResult[3]
+        local requiredRaces = queryResult[4]
+        local requiredClasses = queryResult[5]
+
+        if _IsQuestAvailableToPlayer(requiredRaces, requiredClasses) then
```

Comment:
To (fully?) fix the "Factions without quests are showing up" issue, we should move the filtering of hidden quests to this place.

So the check that is currently done further below when iterating `sortedQuestByLevel` needs to happen here and is then no longer required below.